### ISSUE TITLE
Inheriting documentation including dynamic dots in decorator functions

### ISSUE THREAD

**Issue by robchallen** [r-lib/roxygen2#1670]
I've written a decorator for `dplyr` functions that does some additional work before handing off to `dplyr`. I want to inherit the documentation from `dplyr`, including any documentation of dynamic dots. It seems this is not possible as `@inherit` or `@inheritParams` alone will not import named `dplyr` parameters that are not present in my decorator, and `@inheritDotParams` will not import `dplyr` documentation of `...` as a parameter. Minimal example as below:

```R
#' Function to be inherited from
#'
#' I would like parameters `a`, `b` and `...` to be documented in decorator functions below.
#'
#' @param a param one in function one
#' @param b param two in function one
#' @param ... important documentation of name value parameters in function one.
#'
#' @return stuff
#' @export
#'
#' @examples
#' print("hi")
one = function (a, b, ...) {

}


#' Decorator function calling function one
#'
#' This does document `a` and `b` from function one, but unfortunately this does
#' not document parameter `...` from function one.
#'
#' @param x param one in function two.
#'
#' @inherit one
#' @inheritDotParams one
two = function(x, ...) {
  # do something with x
  one(...)
}


#' Another decorator function calling function one
#'
#' This does document `...` from function one, but unfortunately does not
#' document parameters `a` and `b`
#'
#' @param y param one in function three.
#' @inherit one
three = function(y, ...) {
  # do something with y
  one(...)
}
```

N.B. roxygen2 version 7.3.1
Is there any way to inherit everything? I can see there could be potential recursion issues.

**Comment by hadley** [comment]
Yeah, there's no way to do this sorry, and I can't quite figure out if it's even possible. It does feel like roxygen2 needs better tools for reusing documentation in various ways, but I'm increasingly sceptical that `@inheritParams` and friends are the right overall approach.

**Comment by robchallen** [comment]
I did get it to work on a fork so that '...' parameters accepted by
downstream functions can be documented in the upstream caller. In my view
it makes documentation more transparent but it can get quite large.

If it is something you consider a good idea I can put together a pull
request which adds a @inheritAllDotParams tag that rolls up the
documentation.

WDYT?
